src/main/pileup.js

```diff
@@ -80,6 +80,37 @@ function create(elOrId: string|Element, params: PileupParams): Pileup {
       ReactDOM.render(<Root referenceSource={referenceTrack.source}
                             tracks={vizTracks}
                             initialRange={params.range} />, el);
+
+  //if the element doesn't belong to document DOM observe DOM to detect
+  //when it's attached
+  var observer = null;
+  if (!document.contains(el)) {
+    observer = new MutationObserver(function(mutations) {
+      mutations.forEach(function(mutation) {
+        if (mutation.type === 'childList') {
+          var added= false;
+          for (var i=0; i<mutation.addedNodes.length;i++) {
+            if (mutation.addedNodes[i]===el) {
+              added = true;
+            }
+          }
+          if (added) {
+            if (reactElement) {
+              reactElement.setState({updateSize:true});
+            } else {
+              throw 'ReactElement was not initialized properly';
+            }
+          }
+        }
+      });
+    });
+    // configuration of the observer:
+    var config = { attributes: true, childList: true, characterData: true, subtree: true };
+
+    // start observing document
+    observer.observe(document, config);
```

Comment:
I also don't have any experience. I found an article here:
https://hacks.mozilla.org/2012/05/dom-mutationobserver-reacting-to-dom-changes-without-killing-browser-performance/
They mention it as the proper way to do it when performance is important (I know it's 4 years old, but couldn't find better way to do that).

I was thinking about watching proper elements, but I couldn't find any better solution. My first idea was quite opposit - watch our node element if it's parent has changed, but I couldn't make it work (even though that MutationObserver in theoury allows to observe modification of node attributes - maybe parent is not treated as an attribute...).
This is why I added this observer only when it's not in DOM. Maybe we should disconnect observer when element is attached to DOM.